common/errors.go

```
@@ -118,6 +121,41 @@ func newAPIErrorFromPostgreSQLError(err error) *api_service_protos.TError {
 	return nil
 }
 
+func newAPIErrorFromOracleError(err error) *api_service_protos.TError {
+	// go-ora error mapping:
+	// https://github.com/sijms/go-ora/blob
+	// 	/78d53fdf18c31d74e7fc9e0ebe49ee1c6af0abda/v2/network/oracle_error.go#L20-L57
+	var status ydb_proto.StatusIds_StatusCode
+
+	// TODO: remove this and extract OracleError somehow
+	//       errors.As() does not work with go_ora.OracleError because it does not implement Error interface
+	errorText := err.Error()
+
+	match := oracleRegex.FindStringSubmatch(errorText)
+
+	if len(match) != 2 {
+		return nil
+	}
+
+	tmp, _ := strconv.ParseUint(match[1], 10, 16)
```

Здесь я бы не игнорировал ошибку, а тоже кинул бы панику. Let it crash